services/src/handlers/workflows.rs

```diff
@@ -1450,4 +1510,144 @@ mod tests {
             result.as_slice()
         );
     }
+
+    #[cfg(feature = "xgboost")]
+    #[tokio::test]
+    #[serial]
+    async fn ml_model_from_workflow_task_success() {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>().unwrap();
+        let cfg_backup = cfg.model_defs_path;
+
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let tmp_path = tmp_dir.path();
+        std::fs::create_dir_all(tmp_path.join("pro/ml/xgboost")).unwrap();
+
+        let temp_ml_path = tmp_path.join("pro/ml").to_str().unwrap().to_string();
+
+        set_config("machinelearning.model_defs_path", temp_ml_path).unwrap();
+
+        let exe_ctx_tiling_spec = TilingSpecification {
+            origin_coordinate: (0., 0.).into(),
+            tile_size_in_pixels: GridShape::new([4, 2]),
+        };
+
+        let ctx = InMemoryContext::new_with_context_spec(
+            exe_ctx_tiling_spec,
+            TestDefault::test_default(),
+        );
+
+        let session_id = ctx.default_session_ref().await.id();
+
+        let src_a = generate_raster_test_data_band_helper(vec![1, 2, 3, 4, 5, 6, 7, 8]);
+        let src_b = generate_raster_test_data_band_helper(vec![9, 10, 11, 12, 13, 14, 15, 16]);
+        let src_target = generate_raster_test_data_band_helper(vec![0, 1, 2, 2, 2, 1, 0, 0]);
+
+        let mut training_config: HashMap<String, String> = HashMap::new();
```

Comment:
would it make sense to create a struct with the required parameters and use it in the parameters of the operator?

Reply:
This is already part of the `xgboost-rs` crate. Maybe I can just use the provided functionallity from there with a little bit of refactoring.